src/steps/project/index.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  createDirectRelationship,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+} from '@jupiterone/integration-sdk-core';
+
+import { createAPIClient } from '../../client';
+import { IntegrationConfig } from '../../config';
+
+import { Entities, Relationships, Steps } from '../constants';
+import { createProjectEntity } from './converter';
+
+export async function fetchProjects({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await jobState.iterateEntities(
+    { _type: Entities.CONFIGURATION._type },
+    async (pipelineEntity) => {
+      const pipelineDetail = await apiClient.fetchPipelineDetails(
+        pipelineEntity.id as string,
+      );
+
+      const project = await apiClient.fetchProjectDetails(
+        pipelineDetail.project_slug,
+      );
+
+      const isProjectEntity = await jobState.findEntity(
+        `circleci_project:${project.id}`,
+      );
+
```

Comment:
The pipeline (fetched in pipeline steps) already has property `project_slug`. Because we're not actually making use of it in pipeline converter, it gets stored as rawData behind the scenes.

So here, we can just grab this resource's rawData and access the underlying `project_slug` that way, no need to make another HTTP request.

```suggestion
export async function fetchProjects({
  instance,
  jobState,
  logger,
}: IntegrationStepExecutionContext<IntegrationConfig>) {
  const apiClient = createAPIClient(instance.config);

  await jobState.iterateEntities(
    { _type: Entities.CONFIGURATION._type },
    async (pipelineEntity) => {
      
      const pipeline = getRawData<CircleciPipeline>(pipelineEntity);

      if (!pipeline) {
        logger.warn(
          { _key: pipelineEntity._key },
          'Could not get raw data for pipeline entity',
        );
        return;
      }

      const project = await apiClient.fetchProjectDetails(
        pipeline.project_slug,
      );

      const isProjectEntity = await jobState.findEntity(
        `circleci_project:${project.id}`,
      );

```